### PR TITLE
Added fixes for the mode checkers

### PR DESCRIPTION
**Pull request by AlexDM0**
Addresses issues with the mode checkers.
 - Add waitForMode(setup / normal / dfu), throw an error on timeout.
 - Add getMode(), throw an error when there was no scan before timeout.

### CODE REVIEW

File: crownstone_ble/core/CrownstoneBle.py [hunk 1]
```diff
@@ -108,22 +109,105 @@ async def getCrownstonesByScanning(self, scanDuration=3):
         return gatherer.getCollection()
 
 
-    async def isCrownstoneInSetupMode(self, address, scanDuration=3, waitUntilInRequiredMode=False):
-        _LOGGER.debug(f"isCrownstoneInSetupMode address={address} scanDuration={scanDuration} waitUntilInRequiredMode={waitUntilInRequiredMode}")
-        checker = SetupChecker(address, waitUntilInRequiredMode)
+    async def isCrownstoneInSetupMode(self, address: str, scanDuration=3, waitUntilInSetupMode=False) -> bool:
+        print("isCrownstoneInSetupMode is deprecated. Will be removed in v3. Use either getMode or waitForMode instead.")
```
Comment (vliedel) [inline crownstone_ble/core/CrownstoneBle.py]:
```suggestion
        _LOGGER.warning("isCrownstoneInSetupMode is deprecated. Will be removed in v3. Use either getMode or waitForMode instead.")
```

File: crownstone_ble/core/CrownstoneBle.py [hunk 2]
```diff
@@ -108,22 +109,105 @@ async def getCrownstonesByScanning(self, scanDuration=3):
...
+        result = checker.getResult()
+
+        if result is None:
+            raise CrownstoneBleException(BleError.NO_SCANS_RECEIVED, f'During the {scanDuration} seconds of scanning, no advertisement was received from this address.')
+
+        return result
+
+
+    async def isCrownstoneInNormalMode(self, address, scanDuration=3, waitUntilInNormalMode=False) -> bool:
+        print("isCrownstoneInNormalMode is deprecated. Will be removed in v3. Use either getMode or waitForMode instead.")
```
Comment (vliedel) [inline crownstone_ble/core/CrownstoneBle.py]:
```suggestion
        _LOGGER.warning("isCrownstoneInNormalMode is deprecated. Will be removed in v3. Use either getMode or waitForMode instead.")
```

File: crownstone_ble/core/CrownstoneBle.py [hunk 3]
```diff
@@ -108,22 +109,105 @@ async def getCrownstonesByScanning(self, scanDuration=3):
...
+        await self.ble.scan(duration=scanDuration)
+        BleEventBus.unsubscribe(subscriptionId)
+        result = checker.getResult()
+
+        if result is None:
+            raise CrownstoneBleException(BleError.NO_SCANS_RECEIVED, f'During the {scanDuration} seconds of scanning, no advertisement was received from this address.')
+        if result != requiredMode:
+            raise CrownstoneBleException(BleError.DIFFERENT_MODE_THAN_REQUIRED, f'During the {scanDuration} seconds of scanning, the Crownstone was not in the required mode..')
+
+        return result
```
Comment (vliedel) [inline crownstone_ble/core/CrownstoneBle.py]:
```suggestion
        return
```

**Review by vliedel — COMMENTED** [review]
Looking good.

File: crownstone_ble/core/CrownstoneBle.py (hunk 3 above)
Comment (AlexDM0) [reply crownstone_ble/core/CrownstoneBle.py]: Actually, this return is not required at all.

**Pull request merged**